tasks/compute_comparison.py

```diff
@@ -24,51 +26,89 @@
+        )
+        if not comparison_proxy.has_head_report():
+            comparison.error = CompareCommitError.missing_head_report.value
+            comparison.state = CompareCommitState.error.value
+            log.warning("Comparison doesn't have HEAD report", extra=log_extra)
+            return {"successful": False, "error": "missing_head_report"}
+
+        # At this point we can calculate the patch coverage
+        # Because we have a HEAD report and a base commit to get the diff from
+        patch_totals = async_to_sync(comparison_proxy.get_patch_totals)()
+        comparison.patch_totals = self.minimal_totals(patch_totals)
+
         if not comparison_proxy.has_project_coverage_base_report():
             comparison.error = CompareCommitError.missing_base_report.value
-        elif not comparison_proxy.has_head_report():
-            comparison.error = CompareCommitError.missing_head_report.value
+            log.warning(
+                "Comparison doesn't have BASE report",
+                extra={"base_commit": comparison.base_commit.commitid, **log_extra},
+            )
+            comparison.state = CompareCommitState.error.value
+            return {"successful": False, "error": "missing_base_report"}
         else:
             comparison.error = None
 
-        if comparison.error:
-            comparison.state = CompareCommitState.error.value
-            log.warn("Compute comparison failed, %s", comparison.error, extra=log_extra)
-            return {"successful": False}
         try:
-            with metrics.timer(f"{self.metrics_prefix}.serialize_impacted_files") as tm:
-                impacted_files = self.serialize_impacted_files(comparison_proxy)
+            impacted_files = self.serialize_impacted_files(comparison_proxy)
         except TorngitRateLimitError:
             log.warning(
                 "Unable to compute comparison due to rate limit error",
                 extra=dict(
                     comparison_id=comparison_id, repoid=comparison.compare_commit.repoid
                 ),
             )
-            return {"successful": False}
-        log.info("Files impact calculated", extra=dict(timing_ms=tm.ms, **log_extra))
-        with metrics.timer(f"{self.metrics_prefix}.store_results"):
-            path = self.store_results(comparison, impacted_files)
+            comparison.state = CompareCommitState.error.value
+            return {"successful": False, "error": "torngit_rate_limit"}
+
+        log.info("Files impact calculated", extra=log_extra)
+        path = self.store_results(comparison, impacted_files)
 
         comparison.report_storage_path = path
-        comparison.patch_totals = impacted_files.get("changes_summary").get(
+        calculated_patch_totals = impacted_files.get("changes_summary").get(
             "patch_totals"
         )
+        if calculated_patch_totals != comparison.patch_totals:
+            log.warning(
```

Comment:
under which circumstances would this ever happen?

Reply:
I don't know, to be honest. Hopefully never.

But as I'm switching from that calculation to a copy of the one we use for the notifications I thought it would be prudent to add this just in case.

I also thought it was a good idea to always keep the result that come from the same place